### PR TITLE
Adds vtkImageResample filter to perform linear interpolation slices

### LINKED ISSUE #9277: Add support for vtkImageResample filter
### Description of the enhancement or error report
Adds the vtkImageResample to ImageSampler in MOOSE.

### Rationale for the enhancement or information for reproducing the error
Support for linear interpolation between image slices was requested by a user.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
None, this just is adding features.

### PR DESCRIPTION
**Pull request by aeslaughter**
(closes #9277)

### CODE REVIEW

File: test/tests/functions/image_function/image_resample.i [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  [./u]
+  [../]
+[]
+
+[Functions]
+  [./image_func]
+    type = ImageFunction
+    file_base = stack/test
+    file_suffix = png
+    output_spacing = '1 1 0.5'
```
Comment (aeslaughter) [inline test/tests/functions/image_function/image_resample.i]: I will try to clarify.

Comment (aeslaughter) [inline test/tests/functions/image_function/image_resample.i]: Thanks Mike, now that I am playing with it again I am not sure what is doing. I will figure it out and add some proper documentation.

**Comment by aeslaughter** [comment]
After playing with this a bit, I don't think we need it. The system is already doing what you need using finite elements for the interpolation. If you have one element between slices then you have a linear interpolation built-in. 

So, if you have 10 slices with a spacing of 0.5 in between them, then setup the mesh and/or the "dimension" parameter in the ImageFunction to match the domain of the slices and you should have what you need.

**Comment by tonkmr** [comment]
@aeslaughter where do you dictate that you want a spacing of 0.5 between the slices?

**Comment by aeslaughter** [comment]
You don't you size the mesh, ImageFunction uses this to size itself to sample.

For example, consider an image stack that is 10x10x5, where the 5 is the number of slices. In real life the domain is 1cmx1cmx5cm. The mesh should look like:

```
[Mesh]
  type = GeneratedMesh
  nx = 10 
  ny = 10
  nz = 5
  xmax = 0.01
  ymax = 0.01
  zmax = 0.05
[]
```

Image mesh assumes that the image is the same size as the mesh, it will sample accordingly.

**Pull request closed without merging**